src/Updater.php

```diff
@@ -321,41 +390,54 @@ protected function getFlatKeys(array $nested_array) {
    *   Only if current config is same like old config we are updating.
    * @param array $delete_keys
    *   List of parent keys to remove. @see NestedArray::unsetValue()
+   * @param bool $force
+   *   Force the update.
+   * @param bool $checkOnly
+   *   Check the update status and don't apply the update.
    *
    * @return bool
    *   Returns TRUE if update of configuration was successful.
    */
-  protected function updateConfig($config_name, array $configuration, array $expected_configuration = [], array $delete_keys = []) {
+  protected function updateConfig($config_name, array $configuration, array $expected_configuration = [], array $delete_keys = [], $force = FALSE, $checkOnly = FALSE) {
     $config = $this->configFactory->getEditable($config_name);
 
     $config_data = $config->get();
 
+    // Reset expected_config in case of force flag.
+    if ($force) {
+      $expected_configuration = [];
+    }
+
     // Check that configuration exists before executing update.
     if (empty($config_data)) {
-      return FALSE;
+      return Updater::CONFIG_NOT_FOUND;
     }
 
     // Check if configuration is already in new state.
     $merged_data = NestedArray::mergeDeep($expected_configuration, $configuration);
-    if (empty(DiffArray::diffAssocRecursive($merged_data, $config_data))) {
-      return TRUE;
+    if (!$force && empty(DiffArray::diffAssocRecursive($configuration, $config_data))) {
+      return Updater::CONFIG_ALREADY_APPLIED;
     }
 
-    if (!empty($expected_configuration) && DiffArray::diffAssocRecursive($expected_configuration, $config_data)) {
-      return FALSE;
-    }
+    if (empty($expected_configuration) || !DiffArray::diffAssocRecursive($expected_configuration, $config_data)) {
+      // Delete configuration keys from config.
+      if($checkOnly){
```

Comment:
I'm not sure that I understand the purpose of `$checkOnly`. Is that something like dry-run or?

Additionally, it's not a real representation of an update result, because `$config->save();` could throw an exception of real update execution.